panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsTableInfo.java

```diff
@@ -162,7 +162,10 @@ public void renderGridCellContents(RenderContext ctx, Writer out) throws IOExcep
                                             .at(src, PageFlowUtil.staticResourceUrl("_images/plus.gif"))),
                                     HtmlString.NBSP)
                                     .appendTo(out);
-                            pageConfig.addHandler(spanId, "click", "viewExperimentDetails(this,'" + container.getPath() + "', '" + id + "','" + detailsPage + "')");
+                            pageConfig.addHandler(spanId, "click", "viewExperimentDetails(this,"
+                                    + PageFlowUtil.jsString(container.getPath())
+                                    + ", '" + id + "', "
```

Comment:
Might as well handle them all the same way.

```suggestion
                                    + ", " + PageFlowUtil.jsString(id) + ", "
```

Reply:
'id' is a non-null Integer so does not need to be quoted, I think. 

I added some HTML-escaping in related code.  If you could take another look that would be great.

Reply:
You're right. I somehow thought it was a string.